openpathsampling/features/base.py

```diff
@@ -0,0 +1,231 @@
+from openpathsampling.netcdfplus import DelayedLoader
+from numpydocparse import NumpyDocParser
+
+def has(attr):
+    def _has(func):
+        def inner(self, *args, **kwargs):
+            if hasattr(self, attr) and getattr(self, attr) is not None:
+                return func(self, *args, **kwargs)
+            else:
+                return None
+
+        return inner
+
+    return _has
+
+
+def set_features(*features):
```

Comment:
Could you add some more documentation in here? It is a little hard to tell what all is done by this. The fact that the actual code is *generated* makes it a bit less readable, so more explanation would help me follow it (and remember how it worked when I look at it again in 6 months)